### ISSUE TITLE
is method.isDefault  a java 8 new feature in line 302  

### ISSUE THREAD

**Issue by ylyxf** [jhalterman/typetools#26]
it can't compile in java6

**Comment by mickare** [comment]
Maybe solve it with reflections:

```Java
private static Method METHOD_IS_DEFAULT;
static {
    ...
    Double javaVersion = Double.parseDouble(System.getProperty("java.specification.version", "0"));
    if (javaVersion >= 1.8) {
        ...
        METHOD_IS_DEFAULT = Method.class.getDeclaredMethod("isDefault");
    }
    ...
}

public static boolean isDefault(Method method) {
    if(METHOD_IS_DEFAULT != null) {
        try {
            return (boolean) METHOD_IS_DEFAULT.invoke(method);
        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
        return false;
    }
}

...
if (!isDefault(m) && !Modifier.isStatic(m.getModifiers()) && !m.isBridge()) {
...

```

**Comment by jhalterman** [comment]
TypeTools is [compiled and released as 1.6 bytecode](https://github.com/jhalterman/typetools/blob/master/pom.xml#L122), although admittedly I compile with a 1.8 JDK which is maybe why this problem doesn't occur. You shouldn't hit this problem at runtime since the Java version is checked prior to hitting this line of code.

Is the problem just that you're having trouble compiling TypeTools, not using it? If that's the case, I'd suggest compiling with 1.8 if you can.

**Comment by jhalterman** [comment]
Fixed by #28.
